pkgs/applications/editors/rehex/default.nix

```diff
@@ -0,0 +1,39 @@
+{ capstone
+, fetchFromGitHub
+, jansson
+, lib
+, stdenv
+, wxGTK30
+}:
+
+stdenv.mkDerivation rec {
+  pname = "rehex";
+  version = "0.3.1";
+
+  src = fetchFromGitHub {
+    owner = "solemnwarning";
+    repo = pname;
+    rev = version;
+    sha256 = "1yj9a63j7534mmz8cl1ifg2wmgkxmk6z75jd8lkmc2sfrjbick32";
+  };
+
+  buildInputs = [
+    capstone
+    jansson
+    wxGTK30
+  ];
+
+  makeFlags = [ "prefix=$(out)" ];
+
+  meta = with lib; {
+    description = "Reverse Engineers' Hex Editor";
+    longDescription = ''
+      A cross-platform (Windows, Linux, Mac) hex editor for reverse
+      engineering, and everything else.
+    '';
+    homepage = "https://github.com/solemnwarning/rehex";
+    license = licenses.gpl2;
+    maintainers = with maintainers; [ markus1189 ];
+    platforms = platforms.all;
+  };
+}
```

Comment:
```suggestion
{ stdenv
, fetchFromGitHub
, capstone
, jansson
, wxGTK30
, darwin
, libicns
, wxmac
}:

stdenv.mkDerivation rec {
  pname = "rehex";
  version = "0.3.1";

  src = fetchFromGitHub {
    owner = "solemnwarning";
    repo = pname;
    rev = version;
    sha256 = "1yj9a63j7534mmz8cl1ifg2wmgkxmk6z75jd8lkmc2sfrjbick32";
  };

  patchPhase = ''
    substituteInPlace Makefile.osx --replace 'iconutil -c icns -o $@ $(ICONSET)' 'png2icns $@ $(ICONSET)/icon_16x16.png $(ICONSET)/icon_32x32.png $(ICONSET)/icon_128x128.png $(ICONSET)/icon_256x256.png $(ICONSET)/icon_512x512.png'
  '';

  nativeBuildInputs = stdenv.lib.optionals (stdenv.isDarwin) [ libicns ];

  buildInputs = [ capstone jansson ]
  ++ (stdenv.lib.optionals (!stdenv.isDarwin) [ wxGTK30 ])
  ++ (stdenv.lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [ Carbon Cocoa IOKit wxmac ]));

  makeFlags = [ "prefix=$(out)" ]
  ++ (stdenv.lib.optionals (stdenv.isDarwin) [ "-f Makefile.osx" ]);

  meta = with stdenv.lib; {
    description = "Reverse Engineers' Hex Editor";
    longDescription = ''
      A cross-platform (Windows, Linux, Mac) hex editor for reverse
      engineering, and everything else.
    '';
    homepage = "https://github.com/solemnwarning/rehex";
    license = licenses.gpl2;
    maintainers = with maintainers; [ markus1189 ];
    platforms = platforms.all;
  };
}
```

Removed lib input cause we can use stdenv for that and added darwin building support.

If you don't want to maintain the darwin part you can add me as a maintainer and always ping me about it.

Reply:
Added you as a maintainer and enabled darwin support, thanks! :sunglasses: